Sources/SwiftProtobuf/AnyUnpackError.swift

```diff
@@ -21,6 +21,7 @@
 /// message.  At this time, any error can occur that might have occurred from a
 /// regular decoding operation.  There are also other errors that can occur due
 /// to problems with the `Any` value's structure.
+@available(*, deprecated, message: "This error type has been deprecated and won't be thrown anymore; it has been replaced by `SwiftProtobufError`.")
 public enum AnyUnpackError: Error {
```

Comment:
We have to differentiate here. This is not an API breaking change but might be a semantic breaking change. So just from a SemVer perspective this is fine. However, you are correct that if somebody expects a certain method to throw just a specific set of errors then we might break them. I don't think we ever made that promise nor do we make that promise in any of our other packages. Since Swift uses `any Error` (before typed throws came along) we never made any API level promises to constrain us.

I personally think this change is good and fine and nobody should ever really on specific errors being thrown from an API unless these APIs very explicitly state that.

Reply:
Works for me.  I guess maybe we should better call it out in the comment message and then whenever it is released, we'll have to call it out for that release.

Reply:
fyi - I finally got a chance to go back to read semver - https://semver.org/#spec-item-8 is I think what I was remember. Changing the errors to completely throw different ones is where I could see someone saying we're in the "any backward incompatible changes" space, as existing code that error handled the errors we documented would no longer work.  But I guess until we fully have typed throws in the language, this stays a grey area.

Anyway, resolving this since it sounds like you all are good with this not requiring the bump.